lib/storage: add tracker for time series metric names usage (#8118)
VictoriaMetrics/VictoriaMetrics: docs/changelog/CHANGELOG.md

* FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): look back for recent data for longer time intervals when switching to Table or JSON view. The look back window depends on the `step` setting, which is now set as `end - start` (30m) by default. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/8240).
* FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): preserve user-defined `step` setting when changing the time interval of displayed query. See [this comment](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/8240#issuecomment-2647674065).

> * FEATURE: [vmsingle](https://docs.victoriametrics.com/single-server-victoriametrics/): add new flag `--storage.trackMetricNamesUsage` and new HTTP API - `/api/v1/status/metric_names_usage_stats`. It allows to track metric names usage for ingest and query requests. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4458) for details.

Reviewer: Please fix the API path.

* FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): print full error messages for failed queries on the `Explore Cardinality` page. Before, only response status code was printed. 
* FEATURE: [vmui](https://docs.victoriametrics.com/#vmui): move values representing changes relative to the previous day to a separate column for easier sorting on the `Explore Cardinality` page.
* FEATURE: [MetricsQL](https://docs.victoriametrics.com/metricsql/): support auto-format (prettify) for expressions that use quoted metric or label names. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7703) for details.

> * FEATURE: [vmsingle](https://docs.victoriametrics.com/single-server-victoriametrics/): add new flag `--storage.trackMetricNamesStats` and new HTTP API - `/api/v1/status/metric_names_stats`. It allows to track metric names usage for ingest and query requests. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4458) for details and related [docs](https://docs.victoriametrics.com/#track-ingested-metrics-usage)

Reviewer 2: ```suggestion
* FEATURE: [vmsingle](https://docs.victoriametrics.com/single-server-victoriametrics/): add a new flag `--storage.trackMetricNamesStats` and a new HTTP API - `/api/v1/status/metric_names_stats`. It allows to track how frequent ingested [metric names](https://docs.victoriametrics.com/keyconcepts/#structure-of-a-metric) are used during [querying](https://docs.victoriametrics.com/keyconcepts/#query-data). See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4458) for details and related [docs](https://docs.victoriametrics.com/#track-ingested-metrics-usage)
```
